addons/hr_contract/views/hr_contract_views.xml

```diff
@@ -151,12 +151,32 @@
             <field name="model">hr.contract</field>
             <field name="arch" type="xml">
                 <form string="Current Contract">
-                    <header>
+                    <field name="contracts_count" invisible="1"/>
+                    <header attrs="{'invisible' : [('contracts_count', '=', 0)]}">
                         <field name="state" groups="!hr_contract.group_hr_contract_manager" widget="statusbar"/>
```

Comment:
The field `state` should be added as invisible as it's used below for the calendar_mismatch